xdsl/dialects/snitch_runtime.py

```diff
@@ -106,15 +95,14 @@ def __init__(
         super().__init__(operands=[dst, src, size], result_types=[tx_id])
 
 
-@irdl_op_definition
-class DmaStart2DWideptrOp(SnitchRuntimeBaseOp):
+class DmaStart2DBaseOp(SnitchRuntimeBaseOp, Generic[_T]):
     """
-    Initiate an asynchronous 2D DMA transfer with wide 64-bit pointers
+    Generic base class for starting asynchronous 2D DMA transfers
```

Comment:
If this is generic, I recommend inheriting from the `ABC` class so that they cannot be instantiated.